styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -82,10 +83,15 @@
   private AtomicInteger activeEvents = new AtomicInteger(0);
   private volatile boolean running = true;
 
-  public QueuedStateManager(Time time, Executor workerPool, Storage storage) {
+  public QueuedStateManager(
+      Time time,
+      Executor workerPool,
+      Storage storage,
+      EventConsumer eventConsumer) {
```

Comment:
How come we daisy chain consumption in an ad-hoc way into here? It seems to add an unneeded responsibility to the state manager. The `StateManager` already quacks like an `EventConsumer` (it has a `CompletionStage<Void> receive(Event event)` method). It reacts to events. Just like event consumers do. I would propose a unifying refactor along the lines of `interface StateManager extends EventConsumer`.